### PR TITLE
TDD - Main Nav

### LINKED ISSUE #22: Testing for NavSearch Component

### LINKED ISSUE #23: Testing for ParkBtn Component

### LINKED ISSUE #26: Testing for ParkList Component

### PR DESCRIPTION
**Pull request by jaypeasee**
## Is this a fix or a feature?
feature/testing

## What is the change?
Unit tests the following components:
* `MainNav`
* `ParkList`
* `ParkBtn`
* `NavSearch`

## Where should the reviewer start?
Review the test files for each of the above components.

## How should this be tested?
1. Look to see if I am missing any components that should be tested.
2. Confirm the tests are all passing with the command `npm test src/<folder name>/<component name>.test.js`

## Relevant Tickets
* #36 
* Closes #26
* Closes #23 
* Closes #22 
* #6 

@saraho1123
@jaypeasee
@baileydunning

### CODE REVIEW

File: src/MainNav/MainNav.test.js [hunk 1]
```diff
@@ -15,7 +15,16 @@ describe('MainNav', () => {
...
     expect(screen.getByAltText('National Parkfinder Logo')).toBeInTheDocument()
     expect(screen.getByText('National Parkfinder')).toBeInTheDocument()
   })
-})
\ No newline at end of file
+
+  it('should render a list of park buttons', () => {
+    expect(screen.getByText('Grand Canyon')).toBeInTheDocument()
+    const allButtons = screen.getAllByRole("button")
+    expect(allButtons.length).toBe(56)
```
Comment (baileydunning) [inline src/MainNav/MainNav.test.js:27]: This was a cool solution to ensuring that all of the buttons from our local data are being rendered

File: src/NavSearch/NavSearch.test.js [hunk 2]
```diff
@@ -14,4 +14,14 @@ describe('NavSearch', () => {
...
     searchInput = screen.getByPlaceholderText('search by name')
   })
+
+  it('should render a search bar', () => {
+    expect(searchInput).toBeInTheDocument()
+  })
+
+  it('should call filterButtonsByName with the typed value', () => {
+    userEvent.type(searchInput, 'Grand')
+    expect(filterButtonsByName).toHaveBeenCalledTimes(5)
```
Comment (baileydunning) [inline src/NavSearch/NavSearch.test.js:24]: Good thinking on testing the amount of times the function is called on keyup

**Review by baileydunning — APPROVED** [review]

File: src/MainNav/MainNav.test.js (hunk 1 above)
Comment (saraho1123) [reply src/MainNav/MainNav.test.js:27]: Yes! I agree with Bailey!

File: src/NavSearch/NavSearch.test.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    searchInput = screen.getByPlaceholderText('search by name')
+  })
+
+  it('should render a search bar', () => {
+    expect(searchInput).toBeInTheDocument()
+  })
+
+  it('should call filterButtonsByName with the typed value', () => {
+    userEvent.type(searchInput, 'Grand')
+    expect(filterButtonsByName).toHaveBeenCalledTimes(5)
```
Comment (saraho1123) [inline src/NavSearch/NavSearch.test.js:24]: Nice job figuring this one is matched to how many letters are typed in the input field!

File: src/ParkBtn/ParkBtn.test.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    expect(parkBtn).toBeInTheDocument()
+  })
+
+  it('should call choosePark with the parkCode', () => {
+    userEvent.click(parkBtn)
+    expect(choosePark).toHaveBeenCalledWith("grca")
+    expect(choosePark).toHaveBeenCalledTimes(1)
+  })
+
+  it('should change the pathname on click', () => {
```
Comment (saraho1123) [inline src/ParkBtn/ParkBtn.test.js:36]: Nice! I'm not sure I would have thought of that test!

**Review by saraho1123 — APPROVED** [review]
Nice testing! I think that because we were very organized in how we set up our components, our tests are not as difficult as we originally expected! :)

**Pull request merged**